### ISSUE TITLE
Success file checks in *FileSource should reject globs that include empty dirs

### ISSUE THREAD

**Issue by gerashegalov** [twitter/scalding#1474]
This is a follow up issue to https://github.com/twitter/scalding/pull/1470/ as suggested by @ianoc for a wider discussion

Currently we allow the scenarios described in FileSource:
```
pathIsGood should still be considered just a best-effort test. As an illustration the following
layout with an in-flight job is accepted for the glob dir*/*:
   dir1/_temporary
   dir2/file1
   dir2/_SUCCESS

Similarly if dir1 is physically empty pathIsGood is still true for dir*/* above
On the other hand it will reject an empty output directory of a finished job:
  dir1/_SUCCESS
```

**Comment by isnotinvain** [comment]
I think we should also change the behavior to handle dirs that exist, have a success file, but no data. This seems like a pretty clear indication that this directory is valid and ready to tell you that in fact there is no data here. Reading one of these dirs should not be an error it should provide an empty typed pipe.

@chandler is interested in this behavior

**Comment by gerashegalov** [comment]
@isnotinvain Yes that's what #1470 originally did as a side effect. @ianoc suggested that we do this part of the change as a separate issue here.

**Comment by isnotinvain** [comment]
Yeah. We may even want to keep both behaviors around (we can have two different mixins I guess). But yeah I think the change described here is something we should have (seems to make more sense to me).

**Comment by isnotinvain** [comment]
Also worth noting so I don't forget: even if scalding accepts a dir that only contains a successfile, hadoop's FileInputFormat will not. So part of this logic will need to filter out these paths before passing them along to FileInputFormat

**Comment by gerashegalov** [comment]
FIF.getSplits should return an empty split array in this case. If it does not work, we can address it in Hadoop.
